src/commands/api-mesh/log-get.js

```diff
@@ -0,0 +1,90 @@
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+const { Command } = require('@oclif/core');
+const logger = require('../../classes/logger');
+const { initSdk, initRequestId } = require('../../helpers');
+const { ignoreCacheFlag } = require('../../utils');
+const { getMeshId, getLogsByRayId } = require('../../lib/devConsole');
+require('dotenv').config();
+
+class FetchLogsCommand extends Command {
+	static args = [{ name: 'rayId', required: true, description: 'to fetch the log ' }];
+	static flags = {
+		ignoreCache: ignoreCacheFlag,
+	};
+
+	async run() {
+		await initRequestId();
+
+		logger.info(`RequestId: ${global.requestId}`);
+
+		const { args, flags } = await this.parse(FetchLogsCommand);
+
+		const ignoreCache = flags.ignoreCache;
+		const rayId = args.rayId;
+
+		const { imsOrgId, projectId, workspaceId } = await initSdk({
+			ignoreCache,
+		});
+
+		let meshId = null;
+		try {
+			meshId = await getMeshId(imsOrgId, projectId, workspaceId, meshId);
+			if (!meshId) {
+				throw new Error('MeshIdNotFound');
+			}
+		} catch (err) {
+			this.error(
+				`Unable to get mesh ID. Please check the details and try again. RequestId: ${global.requestId}`,
+			);
+		}
+
+		if (meshId) {
+			try {
+				const meshLog = await getLogsByRayId(imsOrgId, projectId, workspaceId, meshId, rayId);
+				if (meshLog) {
+					this.log('EventTimestampMs : %s', meshLog.eventTimestampMs);
+					this.log('Exceptions : %s', meshLog.exceptions);
+					this.log('Logs : %s', meshLog.logs);
+					this.log('Outcome : %s', meshLog.outcome);
+					this.log('MeshId : %s', meshLog.meshId);
+					this.log('RayId : %s', meshLog.rayId);
+					this.log('MeshUrl : %s', meshLog.url);
+					this.log('RequestMethod : %s', meshLog.requestMethod);
+					this.log('RequestStatus : %s', meshLog.responseStatus);
+				}
+			} catch (error) {
+				if (error.message === 'LogNotFound') {
+					this.error(
+						`No logs found for RayID ${rayId}. Check the RayID and try again. RequestId: ${global.requestId}. Alternatively, you can use the following command to get all logs for a 30 minute time period: \naio api-mesh log-get-bulk --startTime YYYY-MM-DDTHH:MM:SSZ --endTime YYYY-MM-DDTHH:MM:SSZ --filename mesh_logs.csv`,
+					);
+				} else if (error.message === 'ServerError') {
+					this.error(
+						`Server error while fetching logs for RayId ${rayId}. Please try again later. RequestId: ${global.requestId}`,
+					);
+				} else {
+					this.error(
+						`Unable to get mesh logs. Please check the details and try again. If the error persists please contact support. RequestId: ${global.requestId}`,
```

Comment:
How does this error state get triggered?

Reply:
I added that initially to handle case where meshId was not found. However, since implementation changed i added  getMeshId function which already handles this scenario, i'll have to remove it, i forgot to do that.